docs/decisions/0001-use-rust-as-developing-language.md

```diff
@@ -0,0 +1,58 @@
+---
+# These are optional metadata elements. Feel free to remove any of them.
+status: proposed
+date: 2025-03-18
+decision-makers:
+  - '@luca-c-xcv'
+  - '@feed3r'
+  - '@giubacc'
+---
+
+# Use Rust as developing language
+
+## Context and Problem Statement
+
+**wg-bridge** is currently developed in bash, but this language is less
+flexible, less robust, and difficult to maintain for a complex tool.
+Moreover, Bash is not designed for writing encapsulated code, which is becoming
+a necessary requirement as development progresses.
+
+<!-- This is an optional element. Feel free to remove. -->
+## Decision Drivers
+
+* Rust is more flexible.
+* Rust is more robust.
+* Rust is designed for writing complex code.
+* Rust is compiled and does not require an interpreter in the user's
+  environment.
+* Rust is more performant.
+
+## Considered Options
+
+* Python
+* Go
+* Perl
+
+## Decision Outcome
+
+The most notable feature is the compiled language.
+Other notable features are reliability, performance.
+
+<!-- This is an optional element. Feel free to remove. -->
+## Pros and Cons of the Options
```

Comment:
A con for Rust is that it requires a significant cognitive load by us (we have to learn it :)

Reply:
I agree, but I think that this can be one of the most significant reasons: to be honest I think that part of the scope of this PR is to have a nice toy project that we can use as a playground to learn Rust. 

I think that from a programming point of view, this project is not so complicated, and sufficiently adjustable so that we can implement something not complicated (launch some commands, read and write some status files) with the possibility to develop a real world scenario application.

Reply:
ok :smile: , so if learning Rust is one of the major drivers here we should add it to the ADR no? :wink: 
(I agree that this is a main driver and this is a perfectly legitimate one, so add it!)